Assignment_2/code/Problem-4.Rmd

```diff
@@ -0,0 +1,183 @@
+---
+title: "Modelling insurance claims"
+subtitle: "Problem 4"
+output: pdf_document
+---
+
+```{r setup, include=FALSE} 
+knitr::opts_chunk$set(warning = FALSE, message = FALSE) 
+```
+
+$\underline{\textbf{Problem solver:}} \text{ Rohan Karthikeyan}$
+
+$\underline{\textbf{Reviewer:}} \text{ Adarsha Mondal}$
+
+Let's load the dataset and plot a graph of the concerned variables:
+```{r}
+library(MASS)
+plot(Insurance$Holders, Insurance$Claims,
+     xlab = 'Holders', ylab='Claims',
+     pch=20)
+grid()
+```
+
+\newpage
+
+## Model 1: Linear regression with normally distributed errors
+
+```{r}
+NLL_Model1 <- function(theta, y, X)
+{
+  beta_0 = theta[1]
+  beta_1 = theta[2]
+  sigma = exp(theta[3])  # variance of error terms
+
+  n = length(data)
+  # -ve log likelihood function of normal distribution
+  l = -sum(dnorm(y, mean = beta_0 + beta_1*X,
+                 sd = sigma, log=T))
+  return(l)
+}
+```
+
+Let's fit this model to the `Insurance` dataset.
+
+```{r}
+theta_initial1 = c(4, 0.15, 0.35)
+fit_1 = optim(theta_initial1, NLL_Model1,
+              y=Insurance$Claims,
+              X=Insurance$Holders,
+              control=list(maxit=1500))
+theta_hat = fit_1$par
+
+beta0_hat = theta_hat[1]
+beta1_hat = theta_hat[2]
+sigma_hat = exp(theta_hat[3])
+
+paste0("Estimated beta0: ", beta0_hat)
+paste0("Estimated beta1: ", beta1_hat)
+paste0("Estimated sigma: ", sigma_hat)
+```
+
+## Model 2: Linear regression with Laplace distributed errors
+
+```{r}
+NLL_Model2 <- function(theta, y, X)
+{
+  beta_0 = theta[1]
+  beta_1 = theta[2]
+  sigma = exp(theta[3])  # variance of error terms
+  
+  n = length(data)
+  # Log likelihood function of Laplace distribution
+  l = -(n*log(2*sigma) + 1/sigma*sum(abs(y - beta_0 - beta_1*X)))
+  return(-l)
+}
+```
+
+Let's fit this model to the `Insurance` dataset.
+
+```{r}
+theta_initial2 = c(4, 0.15, 0.35)
+fit_2 = optim(theta_initial2, NLL_Model2,
+              y=Insurance$Claims,
+              X=Insurance$Holders,
+              control=list(maxit=1500))
+theta_hat = fit_2$par
+
+beta0_hat = theta_hat[1]
+beta1_hat = theta_hat[2]
+# variance of Laplace model is 2 * sigma^2
+sigma_hat = exp(theta_hat[3])/sqrt(2)
+
+paste0("Estimated beta0: ", beta0_hat)
+paste0("Estimated beta1: ", beta1_hat)
+paste0("Estimated sigma: ", sigma_hat)
+```
+
+## Model 3: Linear regression for log-normally distributed data
+
+```{r}
+NLL_Model3 <- function(theta, y, X)
+{
+  beta_0 = theta[1]
+  beta_1 = theta[2]
+  sigma = exp(theta[3])  # variance of error terms
+
+  n = length(data)

```

Comment:
Please, change `data` to `y`, as `data` is a global variable

Reply:
OK.